form.py

```diff
@@ -267,27 +278,30 @@ def review():
                 report.proud_of_accomplishment = proud
                 report.is_final = is_final
                 
-                db.session.add(report)
-
-                try:
-                    sdt = students.query.filter_by(id=getID()).first()
-                    state = sdt.active
-                    if sdt is None:
-                        displayError('user not found in database')
-                    if state == 'midterm':
-                        # check if already submitted
-                        sdt.midterm_done = 1
-                    elif state == 'final':
-                        # check if already submitted
-                        sdt.final_done = 1
-                    else:
-                        displayError('submitting reports not open')
-
-                    db.session.commit()
-                except:
-                    displayError('submission error')
+                gbmodel.db_session.add(report)
+
+        try:
```

Comment:
At some point, I might want to know how this works, but my lack of it won't stop me from approving